src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,112 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.PEOPLE.clear();
+    }
+
+    @Test
+    void register_allParamsValid_ok() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setAge(18);
+        user.setPassword("123456790");
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(RegistrationServiceException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setAge(18);
+        user.setPassword("123456790");
+        assertThrows(RegistrationServiceException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setAge(null);
+        user.setPassword("123456790");
+        assertThrows(RegistrationServiceException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setAge(18);
+        user.setPassword(null);
+        assertThrows(RegistrationServiceException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_invalidLogin_notOk() {
+        User user = new User();
+        user.setLogin("a");
+        user.setAge(18);
+        user.setPassword("123456790");
+        assertThrows(RegistrationServiceException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        User user = new User();
+        user.setLogin("");
+        user.setAge(18);
+        user.setPassword("123456790");
+        assertThrows(RegistrationServiceException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_invalidAge_notOk() {
```

Comment:
Let's add one case when age is negative

Like in common mistakes [Add tests for all possible User’s parameters (null login/password/age, user under 18/18/over 18 years old, negative age, and so on…).](https://mate-academy.github.io/jv-program-common-mistakes/java-core/junit/registration-validation)